src/routing/bootstrap.rs

```diff
@@ -156,12 +160,19 @@ impl<'a> State<'a> {
                     );
                     return Ok((prefix, key, elders));
                 }
-                GetSectionResponse::Redirect(new_bootstrap_addrs) => {
-                    info!(
-                        "Bootstrapping redirected to another set of peers: {:?}",
-                        new_bootstrap_addrs,
-                    );
-                    bootstrap_addrs = new_bootstrap_addrs.to_vec();
+                GetSectionResponse::Redirect(mut new_bootstrap_addrs) => {
+                    // Ignore already used addresses
+                    new_bootstrap_addrs.retain(|addr| !used_addrs.contains(addr));
+
+                    if new_bootstrap_addrs.is_empty() {
+                        debug!("Bootstrapping redirected to the same set of peers we already contacted - ignoring");
```

Comment:
is that expecting `mem::take` to throw an error on an empty value of bootstrap_addrs to break the loop ?

Reply:
No. If `bootstrap_addrs` is empty, we won't send any more requests but we'll still wait for more responses.

Reply:
ok. then there will be high chance that after a `time_out` error, the loop got break, i.e. no more GetSectionResponse come in.
right?

Reply:
We don't implement any timeout in routing for this, so unless the upper layer adds one, this loops might run forever if we never receive a valid `GetSectionResponse`.